src/libinjection_xss.c

```diff
@@ -341,14 +432,20 @@ static attribute_t is_black_attr(const char* s, size_t len)
     }
 
     if (len >= 5) {
-        /* JavaScript on.* */
+
+	/* JavaScript on.* event handlers */
         if ((s[0] == 'o' || s[0] == 'O') && (s[1] == 'n' || s[1] == 'N')) {
-            /* printf("Got JavaScript on- attribute name\n"); */
-            return TYPE_BLACK;
+            black = BLACKATTREVENT;
+            while (black->name != NULL) {
+                if (cstrcasecmp_with_null(black->name, s, strlen(black->name)) == 0) {
```

Comment:
After actually testing this, I'll have to admit this is actually nonsense. The whole construct works as it should if this line is replaced with:

```if (m_strcasestr(s, *eh) != NULL) {```